Add python rosdep for grpcio, grpcio-reflection and parameterized

f-fl0 opened the pull request:
- add `python-grpcio` entry for `ubuntu:focal` (https://pypi.org/project/grpcio/)
- add `python-grpcio-reflection-pip` for `ubuntu` and `debian` (https://pypi.org/project/grpcio-reflection/)
- add `python-parameterized` entry for `ubuntu:focal` (https://pypi.org/project/parameterized/)

f-fl0 committed:
Add python rosdep for grpcio, grpcio-reflection and parameterized

- add python-grpcio entry for ubuntu:focal
- add python-grpcio-reflection-pip for ubuntu and debian
- add python-parameterized entry for ubuntu:focal

nuclearsandwich commented on rosdep/python.yaml:
```diff
@@ -2890,6 +2899,9 @@ python-parameterized:
         packages: [parameterized]
+    focal:
+      pip:
+        packages: [parameterized]
```
According to https://packages.ubuntu.com/focal/python3-parameterized version 0.7.0 is packaged in Ubuntu Focal. Would now be a good time to add a `python3-parameterized` key which uses the system dependency instead of relying on pip keys?

f-fl0 replied:
Do you suggest adding an entry like this instead?
```
python3-parameterized:
  ubuntu:
    focal: [python3-parameterized]
```

tfoote replied:
yes, we should use the debian packaged version if it exists. pip can be used to backfill necessary platforms where the debian package doesn't exist.

nuclearsandwich commented on rosdep/python.yaml:
```diff
@@ -1966,10 +1966,19 @@ python-grpcio:
     bionic:
       pip: [grpcio]
+    focal:
+      pip: [grpcio]
```
Since the default ROS installation on Ubuntu Focal targets python3. I'm wondering if we should create a new key, `python3-grpcio-pip`. It also appears that this package is now available directly from the Ubuntu repositories for Focal https://packages.ubuntu.com/focal/python3-grpcio. Would it make sense to switch away from the pip key entirely?

f-fl0 replied:
I just realized there is already an entry for `python3-grpcio` https://github.com/ros/rosdistro/blob/master/rosdep/python.yaml#L5574. I can add `focal` there if it is better.

clalancette replied:
> I just realized there is already an entry for `python3-grpcio` https://github.com/ros/rosdistro/blob/master/rosdep/python.yaml#L5574. I can add `focal` there if it is better.

That shouldn't be necessary; the wildcard syntax there already covers focal.  So in point of fact, this part of the change shouldn't be necessary at all.

f-fl0 committed:
Address PR comments

- Remove unnecessary entry for grpcio.
- Use packaged version instead of pip for python3-parameterized.
